ddtrace/contrib/internal/asgi/middleware.py

```diff
@@ -241,9 +259,288 @@ async def __call__(self, scope, receive, send):
             )
             tags = _extract_versions_from_scope(scope, self.integration_config)
             span.set_tags(tags)
+            global_root_span = span._local_root or span
+
+            @wraps(receive)
+            async def wrapped_receive():
+                """
+                websocket.message.type
+                websocket.message.length
+                websocket.message.frames
+                """
+
+                if not self.integration_config._trace_asgi_websocket_messages:
+                    return await receive()
+
+                # Create the span when the handler is invoked (when receive() is called)
+                # This measures the time from receive() call to message processing completion
+                recv_span = self.tracer.start_span(
+                    name="websocket.receive",
+                    service=span.service,
+                    resource=f"websocket {scope.get('path', '')}",
+                    span_type="websocket",
+                    child_of=span if not self.integration_config._websocket_messages_separate_traces else None,
+                    activate=True,
+                )
+                try:
+                    message = await receive()
+
+                    if scope["type"] == "websocket" and message["type"] == "websocket.receive":
+                        # Keep the current receive span open to measure processing time
+                        # It will be finished after the application processes the message
+
+                        # end current receive span before
+                        current_receive_span = scope.get("datadog", {}).get("current_receive_span")
```

Comment:
Is this the previous receive span? The current receive span should be recv_span
```suggestion
                        previous_receive_span = scope.get("datadog", {}).get("current_receive_span")
```